File: companyJava/FolhaSalarial.java
package empresa;

import java.io.IOException;

public class FolhaSalarial {
	// Vetor que contera todos os funcionarios.
	private Funcionario funcionarios[] = new Funcionario[100];
	// Inteiro que armazena quantos funcionarios exitem no vetor acima.
	private int qntdFuncionarios = 0;
	
	public static void main(String[] args) throws IOException {
		int opcao = 0;
		FolhaSalarial fs = new FolhaSalarial();
		
		// Enquanto o usuario nao quiser sair do programa...
		while (opcao != 5) {
			// Lendo acao a ser realizada.
			opcao = leOpcao();
			
			switch (opcao) 
			{
			case 1: // Adicionar Gerente.
				// Lendo as informacoes do Gerente.
				System.out.print("Nome: ");
				String nome = EntradaTeclado.leString();
				System.out.print("CPF: ");
				String cpf = EntradaTeclado.leString();
				System.out.print("Salario Base: ");
				double salarioBase = EntradaTeclado.leDouble();
				
				// Caso o CPF seja valido, adicionando o Gerente na folha salarial.
				if (Funcionario.verificaCPF(cpf) == true) {
					Gerente gerente = new Gerente(nome, cpf, salarioBase);
					fs.adicionaFuncionario(gerente);
					
					System.out.println(" ------- Funcionario adicionado com sucesso -------\n");				
				} else {
					System.out.println(" ------- CPF invalido -------\n");
				}
				
				break;
			case 2: // Adicionar Vendedor.
				// Lendo as informacoes do Vendedor.
				System.out.print("Nome: ");
				nome = EntradaTeclado.leString();
				System.out.print("CPF: ");
				cpf = EntradaTeclado.leString();
				System.out.print("Salario Base: ");
				salarioBase = EntradaTeclado.leDouble();
				System.out.print("Comissão: ");
				double comissao = EntradaTeclado.leDouble();
				
				// Caso o CPF seja valido, adicionando o Vendedor na folha salarial.
				if (Funcionario.verificaCPF(cpf) == true) {
					Vendedor vendedor = new Vendedor(nome, cpf, salarioBase, comissao);
					fs.adicionaFuncionario(vendedor);
					
					System.out.println(" ------- Funcionario adicionado com sucesso -------\n");				
				} else {
					System.out.println(" ------- CPF invalido -------\n");
				}
				
				break;
			case 3: // Adicionar Assistente.
				// Lendo as informacoes do Assistente.
				System.out.print("Nome: ");
				nome = EntradaTeclado.leString();
				System.out.print("CPF: ");
				cpf = EntradaTeclado.leString();
				System.out.print("Salario Base: ");
				salarioBase = EntradaTeclado.leDouble();
				
				// Caso o CPF seja valido, adicionando o Assistente na folha salarial.
				if (Funcionario.verificaCPF(cpf) == true) {
					Assistente assistente = new Assistente(nome, cpf, salarioBase);
					fs.adicionaFuncionario(assistente);
					
					System.out.println(" ------- Funcionario adicionado com sucesso -------\n");				
				} else {
					System.out.println(" ------- CPF invalido -------\n");
				}
				
				break;
			case 4:
				fs.geraFolhaSalarial();
				
				break;
			case 5:
				System.out.println(" ------- Saindo do programa -------\n");
				
				break;
			default:
				System.out.println(" ------- Opção invalida -------\n");
			}			
		}
	}
	
	// Imprime todas as acoes que podem ser realizadas e recebe do usuario qual ele deseja fazer.
	private static int leOpcao() throws IOException {
		int opcao = 0;
		
		System.out.println("1) Adicionar gerente");
		System.out.println("2) Adicionar vendedor");
		System.out.println("3) Adicionar assitente");
		System.out.println("4) Gerar folha salarial");
		System.out.println("5) Sair");
		
		System.out.print("Digite a opcao desejada: ");
		opcao = EntradaTeclado.leInt();			
		
		return opcao;
	}
	
	// Coloca um novo funcionario no vetor e atualiza a quantidade de funcionarios.
	private void adicionaFuncionario(Funcionario funcionario) {
		this.funcionarios[this.qntdFuncionarios++] = funcionario;
	}
	
	// Calcula a soma de todos os salarios e imprime-a junto com a quantidade de funcionarios.
	private void geraFolhaSalarial() {
		int valorTotal = 0;
		
		for (int i = 0; i < this.qntdFuncionarios; i++) {
			valorTotal += funcionarios[i].calculaSalario();
		}
		
		System.out.println("Quantidade de funcionarios: " + this.qntdFuncionarios);
		System.out.println("Folha Salarial: R$" + valorTotal + "\n");
	}
}

File: companyPython/FolhaSalarial.py
from Funcionario import Funcionario
from Gerente import Gerente
from Vendedor import Vendedor
from Assistente import Assistente

class FolhaSalarial:
    def __init__(self):
        # Cada folha salarial possui um vetor de funcionarios e um inteiro que diz quantos funcionarios existem.
        self.funcionarios = []
        self.qntdFuncionarios = 0

    # Imprime todas as acoes que podem ser realizadas e recebe do usuario qual ele deseja fazer.
    def leOpcao():
        print("1) Adicionar gerente")
        print("2) Adicionar vendedor")
        print("3) Adicionar assitente")
        print("3) Adicionar assitente")
        print("4) Gerar folha salarial")
        print("5) Sair")

        print("Digite a opcao desejada: ", end = "")
        opcao = int(input())

        return opcao

    # Coloca um novo funcionario no vetor e atualiza a quantidade de funcionarios.
    def adicionaFuncionario(self, funcionario):
        self.funcionarios.append(funcionario)
        self.qntdFuncionarios += 1

    # Calcula a soma de todos os salarios e imprime-a junto com a quantidade de funcionarios.
    def geraFolhaSalarial(self):
        valorTotal = 0

        for i in self.funcionarios:
            valorTotal += i.calculaSalario()

        print("Quantidade de funcionarios: ", self.qntdFuncionarios)
        print("Folha Salarial: R${:.2f}\n".format(valorTotal))

if __name__ == '__main__':
    opcao = 0
    folha = FolhaSalarial()

    # Enquanto o usuario nao quiser sair do programa...
    while (opcao != 5):
        # Lendo acao a ser realizada.
        opcao = FolhaSalarial.leOpcao()

        if (opcao == 1): # Adicionar Gerente
            # Lendo as informacoes do Gerente.
            print("Nome: ", end = "")
            nome = input()
            print("CPF: ", end = "")
            cpf = input()
            print("Salario Base: ", end = "")
            salarioBase = float(input())

            # Caso o CPF seja valido, adicionando o Gerente na folha salarial.
            if (Funcionario.verificaCPF(cpf)):
                gerente = Gerente(nome, cpf, salarioBase)
                folha.adicionaFuncionario(gerente)

                print(" ------- Funcionario Adicionado com sucesso -------\n")
            else:
                print(" ------- CPF invalido -------\n")
        elif (opcao == 2): # Adicionar Vendedor
            # Lendo as informacoes do Vendedor.
            print("Nome: ", end = "")
            nome = input()
            print("CPF: ", end = "")
            cpf = input()
            print("Salario Base: ", end = "")
            salarioBase = float(input())
            print("Comissao: ", end = "")
            comissao = float(input())

            # Caso o CPF seja valido, adicionando o Vendedor na folha salarial.
            if (Funcionario.verificaCPF(cpf)):
                vendedor = Vendedor(nome, cpf, salarioBase, comissao)
                folha.adicionaFuncionario(vendedor)

                print(" ------- Funcionario Adicionado com sucesso -------\n")
            else:
                print(" ------- CPF invalido -------\n")
        elif (opcao == 3): # Adicionar Assistente
            # Lendo as informacoes do Assistente.
            print("Nome: ", end = "")
            nome = input()
            print("CPF: ", end = "")
            cpf = input()
            print("Salario Base: ", end = "")
            salarioBase = float(input())

            # Caso o CPF seja valido, adicionando o Assistente na folha salarial.
            if (Funcionario.verificaCPF(cpf)):
                assistente = Assistente(nome, cpf, salarioBase)
                folha.adicionaFuncionario(assistente)

                print(" ------- Funcionario Adicionado com sucesso -------\n")
            else:
                print(" ------- CPF invalido -------\n")
        elif (opcao == 4):
            folha.geraFolhaSalarial()
        elif (opcao == 5):
            print(" ------- Saindo do programa -------\n");
        else:
            print(" ------- Opção invalida -------\n");